pkg/config/generate.go

```diff
@@ -35,68 +79,128 @@ func selectNodeImage(cs *acsapi.ContainerService, c *Config) {
 }
 
 func selectContainerImagesOrigin(cs *acsapi.ContainerService, c *Config) {
+	// TODO:
+	// Publish tunnel, sync, images
+	c.ImageConfigFormat = "openshift/origin-${component}:${version}"
 	switch cs.Properties.OrchestratorProfile.OpenShiftConfig.OpenShiftVersion {
+	case "3.11":
```

Comment:
can we share the similar pieces between 3.10 and 3.11? e.g. `c.NodeImage = "docker.io/openshift/origin-node:v" + cs.Properties.OrchestratorProfile.OpenShiftConfig.OpenShiftVersion`

Reply:
previously I had a helper which return pattern based what you pass into it:
```
registry/namespace/name:tag
namespace/name:tag
name:tag
tag
``` is it something you think about or what?

Reply:
Or something like `fmt.Springf("%s/%s/%s", defaultRegistry,defaultNamespace, image:tag)` ?

Reply:
Tag is not always will be version. There might be (and will be for sure) when we need to rollback one particular image from the list to the previous version.
I would better do:
```
dt:="v"+cs.Properties.OrchestratorProfile.OpenShiftConfig.OpenShiftVersion
dn:= "openshift3"
dr:="reg.redhat.com"
c.NodeImage = fmt.Sprintf("%s/%s/%s:%s", dr, dn, "node",dt)
// and we swap values where we need to from defaults? 
c.WebConsoleImage = fmt.Sprintf("%s/%s/%s:%s", dr, dn, "console","v3.11.0")
```
?